lib/vf_settokenvalue.cpp

```diff
@@ -550,14 +551,14 @@ namespace ValueFlow
                         setTokenValue(parent, std::move(result), settings);
                     } else if (Token::Match(parent, "%op%")) {
                         if (Token::Match(parent, "%comp%")) {
-                            if (!result.isFloatValue() && !value1.isIntValue() && !value2.isIntValue())
+                            if (!isFloat && !value1.isIntValue() && !value2.isIntValue())
                                 continue;
                         } else {
                             if (value1.isTokValue() || value2.isTokValue())
                                 break;
                         }
                         bool error = false;
-                        if (result.isFloatValue()) {
+                        if (isFloat) {
```

Comment:
This is wrong, this will set the floatvalue even though it might be an INT, which means we set it to 0 if either operand is a float and the parent is not a float.